source/MaterialXGenOsl/OslShaderGenerator.cpp

```diff
@@ -303,6 +303,10 @@ void OslShaderGenerator::emitFinalOutput(Shader& shader) const
     }
 
     string finalResult = outputSocket->connection->variable;
+    if (!outputSocket->channels.empty())
+    {
+        finalResult = _syntax->getSwizzledVariable(finalResult, outputSocket->connection->type, outputSocket->channels, outputSocket->type);
```

Comment:
Is this always a valid possibility for final output? I thought that for OSL we cannot output to float, vec/color2 or vec4/color4 ? which could be possible with the swizzle. And for GLSL we can only allow vec3/vec4.

Reply:
OSL is allowed to output to any type now. For GLSL we only support vec4 but that is handled later, in the last section of this method.